src/categorical_algebra/DataMigrations.jl

```diff
@@ -242,37 +241,67 @@ function migrate(X::FinDomFunctor, F::GlucSchemaMigration)
   FinDomFunctor(mapvals(ob∘first, colimits_of_limits), funcs, tgt_schema)
 end
 
-# Sigma migration
-#################
+""" Abstract type for a data migration functor.
 
-""" Sigma or left pushforward functorial data migration between acsets.
+This allows the user to have an actual model of the theory of 
+functors, once one knows the exact concrete types Dom and Codom
+of the relevant acsets. 
+"""
+abstract type MigrationFunctor{Dom<:ACSet,Codom<:ACSet} <:
```

Comment:
It seems confusing that the top of the migration hierarchy is `AbstractDataMigration` and the top of the migration functor hierarchy is `MigrationFunctor`. Can we make these follow the same pattern? E.g., we could have `AbstractMigration` and `AbstractMigrationFunctor`.